### PR TITLE
Cherry picked automated images

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

File: controllers/hibp.js [hunk 1]
```diff
@@ -36,6 +36,17 @@ async function notify (req, res) {
...
     }
   }
+  // If we don't have a logo of the breach, get the logo.
+  if (!fs.existsSync(`./public/img/logos/${breachAlert.LogoPath}`)) {
+    try {
+      await HIBP.downloadLogo(breachAlert.LogoPath);
+    } catch(err) {
+      // Set LogoPath to missing-logo-icon.png if download fails
+      // LogoPath will reset when the server restarts
+      log.error(`Unable to download logo for ${breachAlert.LogoPath}.`);
```
Comment (pdehaan) [inline controllers/hibp.js]: I think this needs a string `op` as the first param and an object for the second param:

Something like:
```js
log.error("notify", {message: `Unable to download logo for ${breachAlert.LogoPath}`});
```

File: controllers/hibp.js [hunk 2]
```diff
@@ -36,6 +36,17 @@ async function notify (req, res) {
...
+      breachAlert.LogoPath = "missing-logo-icon.png";
+    }
+  }
 
   if (breachAlert.IsSpamList || breachAlert.Domain === "" || breachAlert.IsFabricated || !breachAlert.IsVerified) {
     log.info(`${breachAlert.Name} is fabricated, a spam list, not associated with a website, or unverified. \n Breach Alert not sent.`);
```
Comment (pdehaan) [inline controllers/hibp.js]: Unrelated, but I filed #694 for this curious `log.info()`...

File: hibp.js [hunk 3]
```diff
@@ -168,6 +176,69 @@ const HIBP = {
...
+      if (parseInt(res.headers["content-length"]) > 500000) {
+        throw new Error(`${fileName} is too large. Downloaded Aborted.`);
+      }
+      if (parseInt(res.headers["content-length"]) > 10000) {
+        fileIsBig = true;
+      } else {
+        // Conditions met, open up writeStream
+        const writeStream = fs.createWriteStream(localSrc);
+        res.pipe(writeStream);
+        writeStream.on("error", err => log.error(err));
```
Comment (pdehaan) [inline hibp.js]: I think this `log.error()` may need to be something like:

```js
log.error("download-logo.error", {err});
```

File: hibp.js [hunk 4]
```diff
@@ -168,6 +176,69 @@ const HIBP = {
...
+        writeStream.on("finish", () => {
+          log.info(`${fileName} logo successfully downloaded.`);
```
Comment (pdehaan) [inline hibp.js]: Similarly, maybe:

```js
log.info("download-logo.finish", {message: `${fileName} logo successfully downloaded.`});
```

File: hibp.js [hunk 5]
```diff
@@ -168,6 +176,69 @@ const HIBP = {
...
+    });
+    stream.on("end", async () => {
+      if(fileIsBig) {
+        logoData = Buffer.concat(logoData);
+        try {
+          logoData = await this.minifyLogoData(logoData);
+          await writeFile(localSrc, logoData, "binary");
+          log.info(`${fileName} logo successfully minified and downloaded.`);
+        } catch (err) {
+          log.error(`Error compressing and saving ${fileName}: ${err}`);
```
Comment (pdehaan) [inline hibp.js]:
```js
log.info("download-logo.end", {message: `${fileName} logo successfully minified and downloaded`});
...
log.error("download-logo.end-error", {message: `Error compressing and saving ${fileName}`, err});
```

File: hibp.js [hunk 6]
```diff
@@ -168,6 +176,69 @@ const HIBP = {
...
+        }
+      }
+    });
+    stream.on("error", err => {
+      log.error(`Error downloading ${fileName} image: ${err.message}`);
```
Comment (pdehaan) [inline hibp.js]:
```js
log.error("download-logo.end-error", {message: `Error downloading ${fileName} image: ${err.message}`);
```

**Comment by pdehaan** [comment]
I think I have a basic question about how this works, after looking at the PR again.

Troy discloses a new breach for example.com. Only our production site gets notified and the latest breach data gets loaded. Does this mean only the production site will then fetch the remote logos?

Hypothetically we now have 3 breaches that have been auto-loaded into monitor.firefox.com and we loaded+minified+stored the logos on the production server. But now what happens when we redeploy? Since those logos were never committed to source control, wouldn't they be lost with a new production deploy, unless somebody still commits them to GitHub? Which, if any of that guessing was correct, means that on production, the logos all work perfectly, until we do a new production deploy, at which point the `` handler kicks in, and shows the missing-logo.png style fallback logo for those 3 breaches that were auto-loaded between production deploys.

**Comment by lesleyjanenorton** [comment]
@pdehaan I think the idea is to move all existing images to a Cloud Storage bucket (https://bugzilla.mozilla.org/show_bug.cgi?id=1518173), write future logos to the bucket, and then be done with having to commit new images to GitHub.

**Comment by groovecoder** [comment]
Closing for now, as we're moving this work to our backlog until we have GCP-based storage all set up. Will keep the branch alive for future reference.

**Pull request closed without merging**